<a href="https://colab.research.google.com/github/Veraeze/AmineRegeneration/blob/main/Objective3.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## Step 1: Install Dependencies and Load Dataset

In this step, we install the required Python libraries and load the amine regeneration dataset for Objective 3. The goal is to build a machine learning model to detect early signs of foaming, fouling, and amine degradation.


In [1]:
#  Install dependencies

!pip install pandas matplotlib seaborn scikit-learn

In [2]:
#  Import libraries

import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
from google.colab import drive

In [3]:
#  Mount Google Drive

drive.mount('/content/drive')

Mounted at /content/drive


In [4]:
#  Load dataset

file_path = '/content/drive/MyDrive/MLDatasets/amine_gen_data.csv'
df = pd.read_csv(file_path, skiprows=[1, 2])


In [5]:
#  Preview first few rows

df.head()

State  stripper - Spec Value (Duty)  \
0  Case 1                    10900000.0   
1  Case 2                    10900000.0   
2  Case 3                    10900000.0   
3  Case 4                    10900000.0   
4  Case 5                    10900000.0   

   stripper - Stage Pressure (Condenser)  \
0                                   20.3   
1                                   20.3   
2                                   20.3   
3                                   20.3   
4                                   20.3   

   stripper - Stage Pressure (Reboiler)  DEA6 - Temperature  \
0                                  23.3                 210   
1                                  23.3                 210   
2                                  23.3                 210   
3                                  23.3                 210   
4                                  23.3                 210   

   DEA6 - CO2 Composition(mole %)  DEA6 - H2S Composition(ppm)  \
0                         0.03972                        171.7   
1                         0.03972                        171.7   
2                         0.03972                        171.7   
3                         0.03972                        171.7   
4                         0.03972                        171.7   

   DEA6 - Molar Flow  DEA6 - Acid Gas Temperature  DEA7 - Temperature  \
0               4193                          210               268.3   
1               4193                          210               268.3   
2               4193                          210               268.3   
3               4193                          210               268.3   
4               4193                          210               268.3   

   DEA7 - CO2 Composition(mole %)  DEA7 - H2S Composition(ppm)  \
0                         0.02120                        107.2   
1                         0.02141                        103.4   
2                         0.02149                        106.6   
3                         0.02149                        104.1   
4                         0.02145                        106.6   

   DEA7 - Molar Flow  
0               4192  
1               4192  
2               4192  
3               4192  
4               4192

## Step 2: Exploratory Data Analysis (EDA)

This step involves exploring the dataset to understand patterns, relationships, and potential issues. We begin by:

- Checking the structure and shape of the dataset,
- Viewing sample records to understand the kind of data collected,
- Identifying missing values and data types,
- Generating summary statistics for numerical columns.

These steps help reveal early insights and guide further preprocessing and modeling.

In [6]:
# Check basic structure of the dataset

df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 49915 entries, 0 to 49914
Data columns (total 13 columns):
 #   Column                                 Non-Null Count  Dtype  
---  ------                                 --------------  -----  
 0   State                                  49915 non-null  object 
 1   stripper - Spec Value (Duty)           49915 non-null  float64
 2   stripper - Stage Pressure (Condenser)  49915 non-null  float64
 3   stripper - Stage Pressure (Reboiler)   49915 non-null  float64
 4   DEA6 - Temperature                     49915 non-null  int64  
 5   DEA6 - CO2 Composition(mole %)         49915 non-null  float64
 6   DEA6 - H2S Composition(ppm)            49915 non-null  float64
 7   DEA6 - Molar Flow                      49915 non-null  int64  
 8   DEA6 - Acid Gas Temperature            49915 non-null  int64  
 9   DEA7 - Temperature                     49915 non-null  float64
 10  DEA7 - CO2 Composition(mole %)         49915 non-null  float64
 11  DE

In [7]:
# Display first 5 rows to preview sample data

df.head()

State  stripper - Spec Value (Duty)  \
0  Case 1                    10900000.0   
1  Case 2                    10900000.0   
2  Case 3                    10900000.0   
3  Case 4                    10900000.0   
4  Case 5                    10900000.0   

   stripper - Stage Pressure (Condenser)  \
0                                   20.3   
1                                   20.3   
2                                   20.3   
3                                   20.3   
4                                   20.3   

   stripper - Stage Pressure (Reboiler)  DEA6 - Temperature  \
0                                  23.3                 210   
1                                  23.3                 210   
2                                  23.3                 210   
3                                  23.3                 210   
4                                  23.3                 210   

   DEA6 - CO2 Composition(mole %)  DEA6 - H2S Composition(ppm)  \
0                         0.03972                        171.7   
1                         0.03972                        171.7   
2                         0.03972                        171.7   
3                         0.03972                        171.7   
4                         0.03972                        171.7   

   DEA6 - Molar Flow  DEA6 - Acid Gas Temperature  DEA7 - Temperature  \
0               4193                          210               268.3   
1               4193                          210               268.3   
2               4193                          210               268.3   
3               4193                          210               268.3   
4               4193                          210               268.3   

   DEA7 - CO2 Composition(mole %)  DEA7 - H2S Composition(ppm)  \
0                         0.02120                        107.2   
1                         0.02141                        103.4   
2                         0.02149                        106.6   
3                         0.02149                        104.1   
4                         0.02145                        106.6   

   DEA7 - Molar Flow  
0               4192  
1               4192  
2               4192  
3               4192  
4               4192

In [8]:
# Check for missing values in each column

print("\nMissing Values per Column:")
print(df.isnull().sum())


Missing Values per Column:
State                                    0
stripper - Spec Value (Duty)             0
stripper - Stage Pressure (Condenser)    0
stripper - Stage Pressure (Reboiler)     0
DEA6 - Temperature                       0
DEA6 - CO2 Composition(mole %)           0
DEA6 - H2S Composition(ppm)              0
DEA6 - Molar Flow                        0
DEA6 - Acid Gas Temperature              0
DEA7 - Temperature                       0
DEA7 - CO2 Composition(mole %)           0
DEA7 - H2S Composition(ppm)              0
DEA7 - Molar Flow                        0
dtype: int64


In [9]:
# Summary statistics of numerical columns

df.describe()

stripper - Spec Value (Duty)  stripper - Stage Pressure (Condenser)  \
count                  4.991500e+04                           49915.000000   
mean                   1.591737e+07                              20.304055   
std                    2.877422e+06                               0.234584   
min                    1.090000e+07                              19.000000   
25%                    1.340000e+07                              20.300000   
50%                    1.590000e+07                              20.300000   
75%                    1.840000e+07                              20.300000   
max                    2.090000e+07                              40.000000   

       stripper - Stage Pressure (Reboiler)  DEA6 - Temperature  \
count                          49915.000000        49915.000000   
mean                              23.304055          210.015026   
std                                0.234584            0.946740   
min                               22.000000          200.000000   
25%                               23.300000          210.000000   
50%                               23.300000          210.000000   
75%                               23.300000          210.000000   
max                               43.000000          295.000000   

       DEA6 - CO2 Composition(mole %)  DEA6 - H2S Composition(ppm)  \
count                    4.991500e+04                 4.991500e+04   
mean                     3.972000e-02                 1.717000e+02   
std                      1.069294e-14                 1.128922e-10   
min                      3.972000e-02                 1.717000e+02   
25%                      3.972000e-02                 1.717000e+02   
50%                      3.972000e-02                 1.717000e+02   
75%                      3.972000e-02                 1.717000e+02   
max                      3.972000e-02                 1.717000e+02   

       DEA6 - Molar Flow  DEA6 - Acid Gas Temperature  DEA7 - Temperature  \
count            49915.0                 49915.000000        49915.000000   
mean              4193.0                   210.015026          268.305716   
std                  0.0                     0.946740            0.326383   
min               4193.0                   200.000000          266.200000   
25%               4193.0                   210.000000          268.300000   
50%               4193.0                   210.000000          268.300000   
75%               4193.0                   210.000000          268.300000   
max               4193.0                   295.000000          294.700000   

       DEA7 - CO2 Composition(mole %)  DEA7 - H2S Composition(ppm)  \
count                    49915.000000                 49915.000000   
mean                         0.016057                    84.315394   
std                          0.002608                    10.133323   
min                          0.012290                    69.610000   
25%                          0.013760                    75.420000   
50%                          0.015660                    82.840000   
75%                          0.018140                    92.460000   
max                          0.021530                   107.200000   

       DEA7 - Molar Flow  
count       49915.000000  
mean         4191.240188  
std             0.427202  
min          4191.000000  
25%          4191.000000  
50%          4191.000000  
75%          4191.000000  
max          4192.000000

The dataset contains **13 columns** and **49,915 rows** with no missing values.

- The **State** column is a case identifier.
- All other features are numerical process variables related to stripper pressure, temperature, flow rates, and gas composition.
- Summary statistics show consistent process ranges with no immediate outliers.
- The data is clean and ready for further analysis.